src/app/app.ts

```diff
@@ -150,6 +161,27 @@ class AppImpl implements App {
         return newOKCallResponseWithMarkdown(msg);
     }
 
+    validateSubNameIsUnique = (subName: string): boolean => {
+        const state = this.call.state;
+        const values = this.call.values;
+
+        // if matching subname does not exist in state, subname is unique
+        const subFound = state.find((option: AppSelectOption) => option.label === subName);
+        if (!subFound) {
+            return true;
+        }
+
+        const subOptions = state.filter((option: AppSelectOption) => option.label === subName);
+        const subSelectLabel = values?.[SubscriptionFields.SubSelectName].label;
+
+        // if changing the subName of an existing subscription, ensure new name does not exist
+        if (subSelectLabel !== subName) {
+            return subOptions.length === 0;
+        }
+
+        return subOptions.length <= 1;
```

Comment:
Just to be clear, the first if statement here runs if we _are_ changing the name, and we correctly fail when the name is already taken by a different subscription. The final return statement checks that only the subscription we are editing has that name (though we should never be in a state where that's the case).

Reply:
I changed some variable naming to try and help clear this function up.  

You are correct in how this function operates with one exception.  If someone did use this App (probably just Dylan and me, but you never know) the final return function can occur.  Suppose I created 5 subs before this code is added, all with the same name.  The last return will ensure the user changes the name of the sub to something unique.